Packs/Flashpoint/Integrations/Flashpoint/Flashpoint.py

```diff
@@ -259,13 +435,481 @@ def replace_key(dictionary, new_key, old_key):
+            first_observed_date = first_observed_date.strftime(READABLE_DATE_FORMAT)  # type: ignore
+
+        data = {
+            'FPID': source.get('fpid', ''),
+            'Email': source.get('email', ''),
+            'Breach Source': source.get('breach', {}).get('source'),
+            'Breach Source Type': source.get('breach', {}).get('source_type'),
+            'Password': source.get('password'),
+            'Created Date (UTC)': created_date,
+            'First Observed Date (UTC)': first_observed_date
+        }
+        hr.append(data)
+
+    return tableToMarkdown("Compromised Credential(s)", hr, ['FPID', 'Email', 'Breach Source', 'Breach Source Type',
+                                                             'Password', 'Created Date (UTC)',
+                                                             'First Observed Date (UTC)'], removeNull=True)
+
+
+def remove_duplicate_records(records: List, fetch_type: str, next_run: dict) -> List:
+    """
+    Function to check for duplicate records and remove them from the list
+
+    :param records: List of records
+    :param fetch_type: Type of the records
+    :param next_run: Dictionary to set in last run
+
+    :return: Updated list of alerts
+    """
+    last_run_key = ''
+    id_key = ''
+    if fetch_type == 'Alerts':
+        last_run_key = 'alert_ids'
+        id_key = 'alert_id'
+    elif fetch_type == DEFAULT_FETCH_TYPE:
+        last_run_key = 'hit_ids'
+        id_key = '_id'
+
+    if next_run.get(last_run_key):
+        prev_alert_ids = next_run[last_run_key]
+        records = [i for i in records if i[id_key] not in prev_alert_ids]
+
+    return records
+
+
+def update_alert_body(alert: dict) -> None:
+    """
+    Function to add highlight to keyword text
+
+    :param alert: The alert object
+
+    :return: None
+    """
+    keyword = alert.get("keyword", {}).get("keyword_text", "").replace('\"', "")
+    body = alert.get("source", {}).get("body", {}).get("text/plain")
+    if body:
+        alert["source"]["body"]["text/plain"] = re.sub(keyword, f"<mark>{keyword}</mark>", body, flags=re.IGNORECASE)
+
+
+def prepare_context_from_next_href(links: str) -> Dict:
+    """
+    Function to prepare context from href
+
+    :param links: Link with the arguments
+
+    :return: Context data made from link
+    """
+    arg_split = links.split('?')
+    context = urllib.parse.parse_qs(arg_split[1])
+    return context
+
+
+def prepare_incidents_from_alerts_data(response: dict, next_run: dict, start_time: str) -> Tuple[dict, list]:
+    """
+    Function to prepare incidents from the alerts data
+
+    :param response: Response from the alerts API
+    :param next_run: Dictionary to set in last run
+    :param start_time: Date time saved of the latest alert
+
+    :return: Tuple of dictionary of next run and list of fetched incidents
+    """
+    incidents = []
+    alerts = response.get('data', [])
+
+    alert_ids = [alert['alert_id'] for alert in alerts]
```

Comment:
Is there a scenario in which `alert_id` won't exist? I am asking as in this case an error will be raised. Consider using get

Reply:
@abaumgarten  Since the business logic is to fetch the Alerts from Flashpoint,  we will always be getting `alert_id` in the API response. This is something we confirmed with our customer but let us confirm it one more time.

Reply:
we're fetching Alerts from Flashpoint in which `alert_id` will always be available in the API response - confirmed with the Flashpoint customer.